gnd/src/main/java/com/google/android/gnd/persistence/shared/FeatureMutation.java

```diff
@@ -40,6 +40,17 @@
    */
   public abstract Optional<Point> getNewLocation();
 
+  @Override
+  public abstract Builder toBuilder();
+
+  /** Returns the mutations of type {@link FeatureMutation} contained in the specified list. */
+  public static ImmutableList<FeatureMutation> filter(ImmutableList<Mutation> mutations) {
+    return stream(mutations)
+        .filter(FeatureMutation.class::isInstance)
+        .map(FeatureMutation.class::cast)
```

Comment:
Why do we need to cast here? Does the compiler require this--is it unable to figure out the type in this case without explicit casting? Wouldn't filter(FeatureMutation.class::isInstance) already concertize to a type: ImmutableList<FeatureMutation>?

Reply:
We can treat `FeatureMutation` as the more generic type `Mutation`, but here we're down casting the items, which in Java is always explicit.